From b5f2d9498e8590db4b07ed05c3bf8f2a1077c8a1 Mon Sep 17 00:00:00 2001
From: Marc Lainez <marc.lainez@gmail.com>
Date: Fri, 13 Mar 2026 09:03:53 +0100
Subject: [PATCH 04/11] dt-bindings: power: supply: add Qualcomm PMI632 QGauge
 fuel gauge

Add device tree binding documentation for the QGauge (QG) fuel gauge
block found in the Qualcomm PMI632 PMIC. This hardware provides battery
voltage and current FIFO sampling, open circuit voltage measurement,
and coulomb counting for battery state estimation.

The binding requires a monitored-battery phandle (simple-battery node)
for OCV-to-capacity lookup tables, and optionally accepts an IIO channel
for battery thermistor temperature readings.

Signed-off-by: Marc Lainez <marc.lainez@gmail.com>
---
 .../bindings/power/supply/qcom,pmi632-qg.yaml | 97 +++++++++++++++++++
 1 file changed, 97 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/power/supply/qcom,pmi632-qg.yaml

diff --git a/Documentation/devicetree/bindings/power/supply/qcom,pmi632-qg.yaml b/Documentation/devicetree/bindings/power/supply/qcom,pmi632-qg.yaml
new file mode 100644
index 000000000..dced89eee
--- /dev/null
+++ b/Documentation/devicetree/bindings/power/supply/qcom,pmi632-qg.yaml
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/power/supply/qcom,pmi632-qg.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Qualcomm PMI632 QGauge Fuel Gauge
+
+maintainers:
+  - Marc Lainez <marc.lainez@gmail.com>
+
+description:
+  The QGauge (QG) block found in the Qualcomm PMI632 PMIC provides battery
+  fuel gauging through hardware voltage and current FIFO sampling, open
+  circuit voltage (OCV) measurement, and coulomb counting. The hardware
+  has a multi-state FSM (S1-S7) that automatically transitions between
+  active sampling and low-power sleep states.
+
+allOf:
+  - $ref: power-supply.yaml#
+
+properties:
+  compatible:
+    const: qcom,pmi632-qg
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    items:
+      - description: Battery missing interrupt
+      - description: Battery voltage low interrupt
+      - description: Battery voltage empty interrupt
+      - description: FIFO data update done interrupt
+      - description: Good OCV measured interrupt
+      - description: FSM state change interrupt
+      - description: QG event interrupt
+
+  interrupt-names:
+    items:
+      - const: batt-missing
+      - const: vbat-low
+      - const: vbat-empty
+      - const: fifo-done
+      - const: good-ocv
+      - const: fsm-state-chg
+      - const: event
+
+  io-channels:
+    items:
+      - description: ADC channel for battery thermistor temperature
+
+  io-channel-names:
+    items:
+      - const: batt-therm
+
+  monitored-battery: true
+
+  power-supplies: true
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - interrupt-names
+  - monitored-battery
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/interrupt-controller/irq.h>
+    pmic {
+      #address-cells = <1>;
+      #size-cells = <0>;
+
+      fuel-gauge@4800 {
+        compatible = "qcom,pmi632-qg";
+        reg = <0x4800>;
+        interrupts = <0x2 0x48 0x0 IRQ_TYPE_EDGE_BOTH>,
+                     <0x2 0x48 0x1 IRQ_TYPE_EDGE_BOTH>,
+                     <0x2 0x48 0x2 IRQ_TYPE_EDGE_RISING>,
+                     <0x2 0x48 0x3 IRQ_TYPE_EDGE_RISING>,
+                     <0x2 0x48 0x4 IRQ_TYPE_EDGE_RISING>,
+                     <0x2 0x48 0x5 IRQ_TYPE_EDGE_RISING>,
+                     <0x2 0x48 0x6 IRQ_TYPE_EDGE_RISING>;
+        interrupt-names = "batt-missing",
+                          "vbat-low",
+                          "vbat-empty",
+                          "fifo-done",
+                          "good-ocv",
+                          "fsm-state-chg",
+                          "event";
+
+        monitored-battery = <&battery>;
+      };
+    };
-- 
2.43.0

